app/src/main/java/com/babylon/wallet/android/domain/usecases/PrepareTransactionForAccountDeletionUseCase.kt

```diff
@@ -1,32 +1,44 @@
 package com.babylon.wallet.android.domain.usecases
 
-import com.babylon.wallet.android.data.dapp.IncomingRequestRepository
 import com.babylon.wallet.android.data.dapp.model.TransactionType
+import com.babylon.wallet.android.di.coroutines.IoDispatcher
+import com.babylon.wallet.android.domain.model.messages.TransactionRequest
 import com.babylon.wallet.android.domain.model.transaction.UnvalidatedManifestData
 import com.babylon.wallet.android.domain.model.transaction.prepareInternalTransactionRequest
 import com.radixdlt.sargon.AccountAddress
 import com.radixdlt.sargon.os.SargonOsManager
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 class PrepareTransactionForAccountDeletionUseCase @Inject constructor(
-    private val incomingRequestRepository: IncomingRequestRepository,
-    private val sargonOsManager: SargonOsManager
+    private val sargonOsManager: SargonOsManager,
+    @IoDispatcher private val coroutineDispatcher: CoroutineDispatcher
 ) {
 
     suspend operator fun invoke(
         deletingAccountAddress: AccountAddress,
         accountAddressToTransferResources: AccountAddress? = null
-    ): Result<Unit> = runCatching {
-        val manifest = sargonOsManager.sargonOs.createDeleteAccountManifest(
-            accountAddress = deletingAccountAddress,
-            recipientAccountAddress = accountAddressToTransferResources
-        )
-        val transactionRequest = UnvalidatedManifestData.from(manifest).prepareInternalTransactionRequest(
-            blockUntilCompleted = true,
-            transactionType = TransactionType.DeleteAccount(deletingAccountAddress)
-        )
+    ): Result<Outcome> = withContext(coroutineDispatcher) {
+        runCatching {
+            val outcome = sargonOsManager.sargonOs.createDeleteAccountManifest(
+                accountAddress = deletingAccountAddress,
+                recipientAccountAddress = accountAddressToTransferResources
+            )
+            val transactionRequest = UnvalidatedManifestData.from(outcome.manifest).prepareInternalTransactionRequest(
+                blockUntilCompleted = true,
+                transactionType = TransactionType.DeleteAccount(deletingAccountAddress)
+            )
 
-        incomingRequestRepository.add(transactionRequest)
-        return Result.success(Unit)
+            Outcome(
+                transactionRequest = transactionRequest,
+                hasNonTransferableResources = outcome.nonTransferableResources.isNotEmpty()
+            )
+        }
     }
+
+    data class Outcome(
+        val transactionRequest: TransactionRequest,
+        val hasNonTransferableResources: Boolean
+    )
```

Comment:
Nice that you removed adding to the queue, the job of this use case was to prepare the manifest.